kernel/src/arch/x86_64/syscall.rs

```diff
@@ -386,41 +393,184 @@ fn handle_fileio(
             let flags = arg3;
             let modes = arg4;
             let _r = user_virt_addr_valid(pid, pathname, 0)?;
-            cnrfs::MlnrKernelNode::map_fd(pid, pathname, flags, modes)
+
+            #[cfg(feature = "exokernel")]
+            {
+                let mut pathname_user_ptr = VAddr::from(pathname);
```

Comment:
are we doing the same parsing again in `return cnrfs::MlnrKernelNode::map_fd(pid, pathname, flags, modes);` then maybe we can factor it out into the common path?

Reply:
same for the other calls

Reply:
To make sure I'm understanding, we could create a parse_user_path function (or similarly named) and call that function from both map_fd & the exokernel syscall path?

I think that makes a lot of sense, and I can definitely do that!

Reply:
yes that sounds good to me